Added empty holders for documentaiton and added rust-job template (#12)
Nor2-io/heim: heim-docs/docs/src/content/docs/application/job.mdx

---
title: Job
description: Job-Application - What is an Job-application
date: 2025-12-12
lastmod: 2025-12-12
maxHeadingLevel: 6
sidebar:
  label: Job Applications
  order: 6
---

import {
  FileTree,
  Card,
  CardGrid,
  LinkCard,
  Steps,
} from "@astrojs/starlight/components";

Job application is an Heim-application with a single [Job-trigger](./../config/triggers/#job-trigger) declared. <br />
Job triggers are a work-load that only gets executed on-command, so each execution gets queued up  either by API or through Heim-portal.<br />
This is also the only executable that accepts regular command-line arguments for each execution. <br />
Limited to maximum queue-size and how many executions you are allow per month, see [Organization Limits](./../../organization/limits/).

## Prerequisites

<LinkCard
  title="Rust Job Application"
  href="./../../../guides/examples/rust-job-application/"
/>

<LinkCard
  title="API-Token"
  href="./../../../organization/limits/"
/>

## Queue a Job
There are two way to queue up a Job, either through the Heim-portal or with the help of the API.

### Heim-portal
Go to your job-application in the Heim-portal and navigate to executions.

> Here we click the button **Run New Job** and we get the possibility to trigger a new Job, together with a string-array of arguments if needed.

Reviewer: ```suggestion
Here we click the button **Run New Job** and we get the possibility to trigger a new Job, together with an optional array of string arguments.
```
